### PR TITLE
Add DB models

### PR DESCRIPTION
**Pull request by algobot76**
Close https://github.com/Dara2004/Goi/issues/44

### CODE REVIEW

File: src/model/Deck.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Model } from "@nozbe/watermelondb";
+import { date, field } from "@nozbe/watermelondb/decorators";
```
Comment (john-zou) [inline src/model/Deck.ts:2]:
```suggestion
import { date, field, children } from "@nozbe/watermelondb/decorators";
```

File: src/model/Deck.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+export default class Deck extends Model {
+  static table = TableName.DECKS;
+  static associations: Associations = {
+    [TableName.CARDS]: { type: "has_many", foreignKey: "deck_id" },
+  };
+
+  @field("name") name;
+  @date("created_at") created_at;
+}
```
Comment (john-zou) [inline src/model/Deck.ts:14]:
```suggestion
  @children("cards") cards;
}
```
Might be useful to have

**Comment by john-zou** [comment]
LGTM 👍 , we can add extended queries (https://nozbe.github.io/WatermelonDB/Query.html) to the models later

File: src/model/Deck.ts (hunk 1 above)
Comment (algobot76) [reply src/model/Deck.ts]: Changed inhttps://github.com/Dara2004/Goi/pull/51/commits/bc9449ef21e1d872f37f908ef91f6f0d7b06e7c0

File: src/model/Deck.ts (hunk 2 above)
Comment (algobot76) [reply src/model/Deck.ts:15]: Changed in https://github.com/Dara2004/Goi/pull/51/commits/bc9449ef21e1d872f37f908ef91f6f0d7b06e7c0

**Pull request merged**